crates/rpc_client/src/eth.rs

```diff
@@ -0,0 +1,185 @@
+where
+    D: serde::Deserializer<'de>,
+{
+    let s: &str = serde::Deserialize::deserialize(deserializer)?;
+    Ok(Some(
+        u64::from_str_radix(&s[2..], 16).expect("failed to parse u64"),
+    ))
+}
+
+fn u64_from_hex<'de, D>(deserializer: D) -> Result<u64, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let s: &str = serde::Deserialize::deserialize(deserializer)?;
+    Ok(u64::from_str_radix(&s[2..], 16).expect("failed to parse u64"))
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct Log {
+    pub address: Address,
+    pub topics: Vec<B256>,
+    pub data: Bytes,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub block_hash: Option<B256>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "optional_u64_from_hex"
+    )]
+    pub block_number: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub transaction_hash: Option<B256>,
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "optional_u64_from_hex"
+    )]
+    pub transaction_index: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub log_index: Option<U256>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub transaction_log_index: Option<U256>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub log_type: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub removed: Option<bool>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Default, serde::Deserialize, serde::Serialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionReceipt {
+    pub block_hash: Option<B256>,
+    #[serde(deserialize_with = "optional_u64_from_hex")]
+    pub block_number: Option<u64>,
+    pub contract_address: Option<Address>,
+    pub cumulative_gas_used: U256,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub effective_gas_price: Option<U256>,
+    pub from: Address,
+    pub gas_used: Option<U256>,
+    pub logs: Vec<Log>,
+    pub logs_bloom: Bloom,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub root: Option<B256>,
+    #[serde(deserialize_with = "optional_u64_from_hex")]
+    pub status: Option<u64>,
+    pub to: Option<Address>,
+    pub transaction_hash: B256,
+    #[serde(deserialize_with = "u64_from_hex")]
+    pub transaction_index: u64,
+    #[serde(
+        rename = "type",
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "optional_u64_from_hex"
+    )]
+    pub transaction_type: Option<u64>,
+}
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, serde::Deserialize, serde::Serialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "camelCase")]
+pub struct Block<TX> {
```

Comment:
```suggestion
pub struct Block<TX>
where TX: Debug, Default, Clone, PartialEq, Eq
{
```

Reply:
adopted in https://github.com/NomicFoundation/hardhat/pull/3383/commits/cfd668a4a680937a8aaa41abc13f1fb37213c918